Azure/azure-cli
# [IoT] Error handling of Hubs missing default storage endpoint

**Description**

This PR fixes a bug where a hub without a default storage endpoint cannot be updated with `az iot hub update`, and also shows an error if the user attempts to update file upload settings on a hub missing this required endpoint.

Fixes #20004 

**Testing Guide**

If your hub has no default storage endpoint (`properties.storageEndpoints[$default]`), you should still be able to update other properties of the hub using `az iot hub update [--param or --set]`, however if you attempt to update file-upload endpoint-specific settings (container name, connection string, sas_ttl, identity, etc), you will receive the following error:

`This hub has no default storage endpoint for file upload.`
`Please recreate your default storage endpoint by running "az iot hub update --name {hub_name} --fcs {storage_connection_string} --fc {storage_container_name}"
`


**History Notes**


[IoT] az iot hub update: Adds error handling for file-upload parameters and fixes empty $default storage endpoint errors

---

This checklist is used to make sure that common guidelines for a pull request are followed.
